### PR TITLE
Feature/legend editable

### PR DESCRIPTION
**Pull request by CloudNiner**
This makes the legend styling editable by clicking on it to display the OverlayPropertiesBar.

Added with/height edit properties so that hte user can adjust the legend size when contents start to overflow.

A few additional TODOs:
- Clean up UX related to clicking to display/hide the edit tools
- Move display related code into cartodb.js for use in dynamic maps

Bubble legend isn't consistent with other legends due to the white png mask used to display the bubbles

### CODE REVIEW

**Comment by flibbertigibbet** [comment]
A few things:
- [x] When the legend edit bar first comes up, the numeric fields are empty, and on first interaction with the up/down arrows, they display `NaN` text, which is then difficult to delete/type over.
- [x] Width and height values only display if the legend has been moved to a custom location. When the values first display in that case, they have the 'px' suffix present but cut off in the display until the value is changed.
- [x] Change of font size/typeface/color does not affect the legend titles.
- [x] Background opacity field could use a label.
- [x] Making a formatting change after a legend move will cause the legend to jump to its previous location.

Looks like a good start!

File: lib/assets/javascripts/cartodb/table/overlays/map_overlays.js [hunk 1]
```diff
@@ -49,6 +91,14 @@ cdb.admin.MapOverlays = cdb.core.View.extend({
       return;
     }
 
+    // Set legendStyleModel to initial state of the legend css styles
+    if (this.legend._models && this.legend._models.length) {
```
Comment (flibbertigibbet) [inline lib/assets/javascripts/cartodb/table/overlays/map_overlays.js]: Why not use `getModels`/`count` from cartodb-org/cartodb.js#2 here?

**Comment by flibbertigibbet** [comment]
Moving the legend and reloading the page with this moves the legend location back to a previously saved location.

File: lib/assets/javascripts/cartodb/table/mapview.js [hunk 2]
```diff
@@ -1318,14 +1318,6 @@ cdb.admin.MapTab = cdb.core.View.extend({
     this.mapView.$el.append(this.stackedLegend.render().$el);
     this.addView(this.stackedLegend);
     if (this.overlays) this.overlays.setLegend(this.stackedLegend);
-
-    if (this.legends && this.legends.length) {
```
Comment (flibbertigibbet) [inline lib/assets/javascripts/cartodb/table/mapview.js:1322]: Adding this back I think should fix issues with the legend location inconsistency.

Comment (CloudNiner) [inline lib/assets/javascripts/cartodb/table/mapview.js:1322]: Ah. This code was effectively moved inside the setlegend call below.

File: lib/assets/javascripts/cartodb/table/overlays/map_overlays.js (hunk 1 above)
Comment (CloudNiner) [inline lib/assets/javascripts/cartodb/table/overlays/map_overlays.js]: Derp. This was one of the places I added those helpers for and missed it.

**Comment by CloudNiner** [comment]
Ready for further review, will update this PR with the updated cartodb.js submodule commit and code once that PR is merged.

**Comment by flibbertigibbet** [comment]
Making a format change after a legend drag without first clicking away still causes legend to jump to last saved location.

The reset button is centered in its box when the legend is in the bottom right, but floats to the top left after the legend moves.

**Comment by CloudNiner** [comment]
Ok, I think I cleared both those issues up, pull down the extra commits and give it a shot.

Also added support for static maps export:
bb-static-export

**Comment by flibbertigibbet** [comment]
Looks good! +1 on tests passing.

**Comment by CloudNiner** [comment]
Cool, build passed on the PR but not the push, will rebase down and let it go one more time, but it looks like a transient npm issue atm.

**Comment by flibbertigibbet** [comment]
Ok, that's fine then.

**Comment by javisantana** [comment]
One question, should the legend look the same in the editor and the embed. I don't see legend changes in the map embed, should I?

**Comment by jbranigan** [comment]
The legend changes haven't been ported to cartodb.js yet, so it shows in the editor and in the image export at this point. The vis.json has the legend info, but cartodb.js doesn't know how to interpret it yet.

**Comment by javisantana** [comment]
ok, thanks

**Pull request merged**